examples/contracts/generic_contract/src/generic.rs

```diff
@@ -32,12 +33,16 @@ impl<InstantiateParam, ExecParam, QueryParam, MigrateParam, RetType, FieldType>
     // It's because we have to map unique generics used as they can be used multiple times.
     // If for some reason like here one type would be used in place of two generics either full
     // path or some alias has to be used.
+    //
+    // Sylvia will fail to recognize generic used if their path is different.
+    // F.e. if we this query would return `SvCustomMsg` and we would pass
+    // `sylvia::types::SvCustomMsg` to the `Generic` trait paths would not match.
```

Comment:
I don't understand it, and I don't like this limitation. It should work as in Rust. Could you explain what does it exactly mean?

Reply:
This is because when we check if generic type is used in context of given type it compares their full `path`.
In such case `Msg`, `types::Msg` and `sylvia::types::Msg` are three different types.

Reply:
I don't find this explains everything. `types::Msg` and `sylvia::types::Msg` are obviously not generic parameters - they are coming from a module, so you just should assume they exist and let Rust resolve it. `Msg` is either a generic type or a type available in global scope. Still, its Rust that would handle figuring out the exact type, but for additional codegen you should consider a type being maybe a generic argument only if it has no path besides the ident. If that is the case, you should be able to see the generic parameters of your definition block. If the ident you face is defined as generic, it is generic. Otherwise, it is not generic, and you don't make any further assumptions about that. I need help understanding how scoping makes any deal here. Could you give some very particular examples of what is possible and what is not? And I am not asking about a description of an example, but an example itself - I think some misunderstanding is happening here. We can a call about this if you want as it seems to be not trivial.